### PR TITLE
Rover corrections to slew rate and safety limits  

### PR DESCRIPTION
**Pull request by khancyr**

### CODE REVIEW

**Comment by rmackay9** [comment]
I'm not familiar with the slew limit stuff (yet) but the "set correct safety limit" looks correct.  I think the "safety-pwm" value is the one that is output when the main CPU loses contact with the I/O board.  It makes sense to be using the TRIM value from the output channels and not the input channels.

**Comment by khancyr** [comment]
The slew rate limit the how much change is allow in 1 second : ex : without you can go from 2000us to 1000us (form full speed to 0 speed) in one loop. It put hight constraint on motor or motor control. With slew rate at 50%,  you will go from 2000us to 1000us in 2s and see a nice speed decrease.
The correction here use SRV_Channels::limit_slew_rate function that replace what was done previously but using servo out constraint (between min - max)

File: APMrover2/Steering.cpp [hunk 1]
```diff
@@ -3,18 +3,13 @@
...
-        uint16_t pwm;
-        if (SRV_Channels::get_output_pwm(SRV_Channel::k_throttle, pwm)) {
-            SRV_Channels::set_output_pwm(SRV_Channel::k_throttle, constrain_int16(pwm, last_throttle - temp, last_throttle + temp));
+void Rover::throttle_slew_limit(void)
+{
+    if (g.throttle_slewrate > 0) {
+        SRV_Channels::limit_slew_rate(SRV_Channel::k_throttle, g.throttle_slewrate, G_Dt);
+        if (g.skid_steer_out) {
+            // when skid steering also limit 2nd channel
+            SRV_Channels::limit_slew_rate(SRV_Channel::k_steering, g.throttle_slewrate, G_Dt);
```
Comment (khancyr) [inline APMrover2/Steering.cpp]: mistake here

**Comment by karthikdesai** [comment]
Awesome! I broke 2 of my motors because it used to yaw on the spot just too quickly. The slewrate wasn't really being obeyed. 
Gonna try this patch soon

File: APMrover2/Steering.cpp [hunk 2]
```diff
@@ -288,14 +282,13 @@ void Rover::set_servos(void) {
                 SRV_Channels::set_output_scaled(SRV_Channel::k_steering, 0);
             }
         }
+    }
 
+    if (control_mode != MANUAL && control_mode != LEARNING) {
```
Comment (OXINARF) [inline APMrover2/Steering.cpp]: Why is this if inserted here? The previous else already meets this condition.

**Comment by khancyr** [comment]
closed, was part of https://github.com/ArduPilot/ardupilot/pull/6483

**Pull request closed without merging**